### PR TITLE
CXF:88606 GHA UAT Port Connectivity terra tests

### PR DESCRIPTION
**Pull request by srushti-patl**
- Added workflow for UAT Test suite
- Added terra test for port-connectivity examples

### CODE REVIEW

File: .github/workflows/terratests-uat-suite.yaml
Comment (srushti-patl) [inline .github/workflows/terratests-uat-suite.yaml:1]: I have added cloud-router-2-port-connection secret here, I will remove it in my next commit update.

**Review by thogarty — CHANGES_REQUESTED** [review]
Requested Changes:
1. We don't need the for loops or if statements in the workflow file. You can just do the `echo`s with the full path
2. We only need one testing go file. Don't need separate go files for each test. You can add each test to the same file
3. Remove the explicit "endpoint" statements from the main.tf files in the tests.

**Review by thogarty — CHANGES_REQUESTED** [review]
Good addition! I noticed that we're not adding `sensitive = true` to the variables for auth related to CSPs though. We need to include that so the CSP credentials aren't exposed as well.

File: examples/cloud-router-2-google-connection/variables.tf [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
   type        = string
+  sensitive = true
 }
 variable "equinix_client_secret" {
   description = "Equinix client secret ID (consumer secret), obtained after registering app in the developer platform"
   type        = string
+  sensitive = true
 }
 #Google Provider
 variable "google_region" {
```
Comment (thogarty) [inline examples/cloud-router-2-google-connection/variables.tf:12]: Missing `sensitive = true` on google credentials.

File: examples/port-2-google-connection/variables.tf [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
   type        = string
+  sensitive = true
 }
 variable "equinix_client_secret" {
   description = "Equinix client secret ID (consumer secret), obtained after registering app in the developer platform"
   type        = string
+  sensitive = true
 }
 #Google Provider
 variable "google_region" {
```
Comment (thogarty) [inline examples/port-2-google-connection/variables.tf:12]: Missing `sensitive = true` on goodle credentials.

File: examples/port-2-ibm2-connection/variables.tf [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
   type        = string
+  sensitive = true
 }
 variable "equinix_client_secret" {
   description = "Equinix client secret ID (consumer secret), obtained after registering app in the developer platform"
   type        = string
+  sensitive = true
 }
 
 variable "connection_name" {
```
Comment (thogarty) [inline examples/port-2-ibm2-connection/variables.tf:12]: Missing `sensitive = true` on zSide Auth Key.

**Review by thogarty — CHANGES_REQUESTED** [review]
Seeing some missed items. Additionally, when it is in final review state; add a comment with the results from latest GHA run to verify success.

File: examples/cloud-router-2-google-connection/variables.tf (hunk 1 above)
Comment (srushti-patl) [reply examples/cloud-router-2-google-connection/variables.tf:12]: We don't have any Google credentials for passing secrets directly. Instead, we are passing secrets using file names.

File: examples/port-2-google-connection/variables.tf (hunk 2 above)
Comment (srushti-patl) [reply examples/port-2-google-connection/variables.tf:12]: We don't have any Google credentials for passing secrets directly. Instead, we are passing secrets using file names.

**Comment by srushti-patl** [comment]
Screenshot 2024-03-19 at 2 25 23 PM
All Port Connectivity Test Cases are Passing

**Review by thogarty — APPROVED** [review]
LGTM. Ship it.

**Pull request merged**